discovery/manager.go

```diff
@@ -88,13 +114,17 @@ func NewManager(ctx context.Context, logger log.Logger, options ...func(*Manager
 		logger = log.NewNopLogger()
 	}
 	mgr := &Manager{
-		logger:         logger,
-		syncCh:         make(chan map[string][]*targetgroup.Group),
-		targets:        make(map[poolKey]map[string]*targetgroup.Group),
-		discoverCancel: []context.CancelFunc{},
-		ctx:            ctx,
-		updatert:       5 * time.Second,
-		triggerSend:    make(chan struct{}, 1),
+		logger:      logger,
+		syncCh:      make(chan map[string][]*targetgroup.Group),
+		targets:     make(map[poolKey]map[string]*targetgroup.Group),
+		providers:   newProviders(),
+		ctx:         ctx,
+		updatert:    5 * time.Second,
+		triggerSend: make(chan struct{}, 1),
+		rnd:         rand.New(rand.NewSource(time.Now().UnixNano())),
+		hashOpts: hashstructure.HashOptions{
```

Comment:
Why is this a parameter?

Reply:
hashOpts is scoped to the manager, private field seems a logical place to store it. Setting non-default value in a constructor also feels natural. Am I missing something?